src/main/java/com/sucy/skill/dynamic/target/TargetComponent.java

```diff
@@ -108,6 +110,7 @@ List<LivingEntity> determineTargets(final LivingEntity caster,
     boolean isValidTarget(final LivingEntity caster, final LivingEntity from, final LivingEntity target) {
         if (SkillAPI.getMeta(target, MechanicListener.ARMOR_STAND) != null) return false;
         if (target instanceof TempEntity) return true;
+        if (invulnerable || target.isInvulnerable()) return false;
```

Comment:
This should probably be

`if (!invulnerable && target.isInvulnerable()) return false;`

This would effectively mean that if we _aren't_ supposed to be targeting invulnerable players and they are invulnerable, they aren't a valid target. If invulnerable is true, the other valid target checks go into effect instead since the condition isn't met.